src/main/java/com/deque/axe/android/rules/hierarchy/base/TouchSize.java

```diff
@@ -43,17 +43,32 @@ public String runRule(AxeProps axeProps) {
     final long height = frame.height();
     final long width = frame.width();
 
-    if (dpi <= 0 || height < 0 || width < 0) {
+    final int screenHeight = axeProps.get(AxeProps.Name.SCREEN_HEIGHT, Integer.class);
+    final int screenWidth = axeProps.get(AxeProps.Name.SCREEN_WIDTH, Integer.class);
+
+    if (dpi <= 0 || height < 0 || width < 0
```

Comment:
I don't think these are border cases, I think this logics belongs in `isOffScreen` as this isn't "special circumstances" this is... if you dig into the AOSP... what happens when android doesn't give a shit about a control, because it's far off screen, or because it is scrolled above the screen.

Reply:
Talked in person because this used to exist and serves a different purpose. Extracted to `isRendered` method.